### DISCUSSION TITLE
Self/outer join in query (javascript client)

### DISCUSSION THREAD

**Discussion by joakimgk** [supabase/supabase#12966] [Questions]
Is it possible to "self/outer join" when querying, using the Javascript client? As in, I want to select all rows from table `tableA` matching my user_id, and I want to also get a collection (sub query?) containing any _other_ users who have the same (foreign) key. Something like this,

```
const { data, error } = await supabase
    .from('tableA')
    .select(`
            id,
            user_id,
            tableB ( 
              name,
              item_id
            )
            tableA:item_id (
              user_id
            )
        `)//.eq('user_id', user_id);
```
Here, `tableB` contains more details on the items and there is a foreign key relationship between the tables (on `item_id`). As I understand it, this is how those two tables are joined, on the FK, which is why I don't have to specify any "on" relation between them?

My aim is to get a set of all my items, but also a collection of the _other_ users who have added the same item. Probably, this can be solved either using a view in the database, or simply by joining two queries in javascript, but I wanted to see if this can be done directly through the supabase client?

**Comment by steve-chavez** [comment]
Yes, self joins work. Did you get any error when trying to do it?
